docs/guidance/rayserve-dev-doc.md

```diff
@@ -0,0 +1,128 @@
+# Develop Ray Serve Python scripts on RayCluster
+
+Many RayService issues are related to the Ray Serve Python scripts.
+We strongly recommend that users develop and test their Ray Serve scripts in a RayCluster using the same Ray image before deploying them to a RayService.
+In this tutorial, we will show you how to develop [ray-service.mobilenet.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-service.mobilenet.yaml).
+Please refer to [ray-service.mobilenet.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-service.mobilenet.yaml) and [mobilenet-rayservice.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/mobilenet-rayservice.md) for more details.
+
+
+# Step 1: Install a KubeRay cluster
+
+Follow this [document](../../helm-chart/kuberay-operator/README.md) to install the latest stable KubeRay operator via Helm repository.
+
+# Step 2: Create a RayCluster CR
+
+```sh
+helm install raycluster kuberay/ray-cluster --version 0.6.0-rc.0
+```
+
+# Step 3: Log in to the head Pod
+
+```sh
+export HEAD_POD=$(kubectl get pods --selector=ray.io/node-type=head -o custom-columns=POD:metadata.name --no-headers)
+kubectl exec -it $HEAD_POD -- bash
+```
+
+# Step 4: Prepare your Ray Serve Python scripts and run the Ray Serve application
+
+```sh
+# Execute the following command in the head Pod
+git clone https://github.com/ray-project/serve_config_examples.git
+cd serve_config_examples
+
+# Try to launch the Ray Serve application
+serve run mobilenet.mobilenet:app
+# [Error message]
+#     from tensorflow.keras.preprocessing import image
+# ModuleNotFoundError: No module named 'tensorflow'
+```
+
+* `serve run mobilenet.mobilenet:app`: The first `mobilenet` is the name of the directory in the `serve_config_examples/`,
+the second `mobilenet` is the name of the Python file in the directory `mobilenet/`, and `app` is the name of the variable representing Ray Serve application within the Python file. See the section "import_path" in [rayservice-troubleshooting.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayservice-troubleshooting.md) for more details.
+
+# Step 5: Change the Ray image from `rayproject/ray:${RAY_VERSION}` to `rayproject/ray-ml:${RAY_VERSION}`
+
+```sh
+# Uninstall RayCluster
+helm uninstall raycluster
+
+# Install the RayCluster CR with the Ray image `rayproject/ray-ml:${RAY_VERSION}`
+helm install raycluster kuberay/ray-cluster --version 0.6.0-rc.0 --set image.repository=rayproject/ray-ml
+```
+
+The error message in Step 4 indicates that the Ray image `rayproject/ray:${RAY_VERSION}` does not have the TensorFlow package.
+Due to the significant size of TensorFlow, we have opted to use an image with TensorFlow as the base instead of installing it within the Ray runtime environment.
```

Comment:
Can we link to the ray runtime environment documentation here?

Reply:
Updated: https://github.com/ray-project/kuberay/pull/1250/commits/638d7bc493aa446013ef11639076fc56d5cd7f91